### PR TITLE
Bomb system

### LINKED ISSUE #12: BombExploderSystem takes timeOutEvent and informs hits targets.

### PR DESCRIPTION
**Pull request by JacopoG**
Related to #12. Highlights:
- When the bomb explode I replace its texture with the explosion texture.
- I added a boolean field `isExploded` in the `ExploderComponent`. This way the collider system should be able to detect the collision between the explosion and the player.

I guess the collider system should inform the health system about the hit player, because it knows which entities are near each other. Give me feedback!

### CODE REVIEW

**Comment by DeimosA** [comment]
Shouldn't the timer system send an event when the time is out?

Also GameWorld.kt doesn't exist anymore so you'll get some merge issues there

**Comment by JacopoG** [comment]
Yes, probably an event make more sense here. In the first place I implemented it this way because sometimes I'm unsure if the communications between systems should use components or events. Maybe I can implement something similar to the event handling logic in #38, probably here it is even simpler. @DeimosA sounds good?

**Review by DeimosA — CHANGES_REQUESTED** [review]
Gameworld.kt needs to be properly integrated with master code

**Comment by DeimosA** [comment]
localplayerid is now gone in master code, and playercomponent has a field islocalplayer.
movementcomponent has been repurposed to playercomponent

File: core/src/no/group15/playmagic/ecs/systems/MovementSystem.kt [hunk 1]
```diff
@@ -116,4 +117,17 @@ class MovementSystem(
 			}
 		}
 	}
+
+	fun localPlayerPosition(): Vector2 {
```
Comment (DeimosA) [inline core/src/no/group15/playmagic/ecs/systems/MovementSystem.kt]: this code is outdated

File: core/src/no/group15/playmagic/ecs/systems/BombExploderSystem.kt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+
+	override fun receive(command: Command) {
+		when (command) {
+			is DropBombCommand -> {
+				val bomb = EntityFactory.makeEntity(assetManager, engine as PooledEngine, EntityFactory.Type.BOMB)
+				bomb[timer]!!.timeLeft = 3f
+
+				// position
+				val playerPos = engine.getSystem(MovementSystem::class.java).localPlayerPosition()
```
Comment (DeimosA) [inline core/src/no/group15/playmagic/ecs/systems/BombExploderSystem.kt]: do a loop here with PlayerComponent family to find local player

**Review by DeimosA — CHANGES_REQUESTED** [review]

File: core/src/no/group15/playmagic/ecs/systems/BombExploderSystem.kt (hunk 2 above)
Comment (JacopoG) [reply core/src/no/group15/playmagic/ecs/systems/BombExploderSystem.kt]: done!

File: core/src/no/group15/playmagic/ecs/systems/MovementSystem.kt (hunk 1 above)
Comment (JacopoG) [reply core/src/no/group15/playmagic/ecs/systems/MovementSystem.kt]: Removed!

**Comment by JacopoG** [comment]
@DeimosA @eldarht now it should be ready for merge!

**Review by DeimosA — APPROVED** [review]

**Review by eldarht — APPROVED** [review]

**Pull request merged**